### ISSUE TITLE
Installing Jasy 0.7.2 with pip-3.2 is missing required dependencies

### ISSUE THREAD

**Issue by Zenwolf** [zynga/jasy#37]
When I install jasy 0.7.2 from pip-3.2, it does not install any required dependencies. It will report errors if the following are not installed:
- polib
- Pygments
- misaka

These should be automatically installed when I install jasy if I do not already have them installed. The build still works without polib but not without misaka and Pygments.

polib error:
"Polib is no installed. Polib is needed to extract translation text from PO files!"

Markdown error:
"Missing Markdown feature to convert comments to HTML."

**Comment by Zenwolf** [comment]
I encounter the markdown error in both a "source" task and a doc "api" task. It seems like it should only fail in the doc one.

**Comment by Zenwolf** [comment]
Also, I am not sure what you would like to do about "Python PIL", but it appears to be required to use the sprite features. I am not using the new sprites, so it doesn't block me, although I see the warning every time:

"Python PIL is not installed. PIL is used to create sprite images."

**Comment by swernerx** [comment]
The new way is make all deps optional. The messages you see are just warnings. You have to install specific modules for features you want to use:
- generating api data => misaka
- generating sprite images => PIL
- translation via po files => polib

You can typically generate source/build tasks without any of these. This is why we have removed these from the installation requirements. Jasy installs without any deps right now and that should work - hopefully.

**Comment by Zenwolf** [comment]
Ok, but I just want to make sure you know that without the Pygments and misaka a basic source task fails and the build stops.

```
$ pip-3.2 uninstall Pygments
...
$ jasy source
...
>>> STORING KERNEL...
-------------------------------------------------------------------------------
Detecting dependencies...
- Processing class core.Env [fields]...
  - Missing Markdown feature to convert comments to HTML.
$
```

**Comment by swernerx** [comment]
Oh interesting. That is unexpected. I'll have a look at it! Thanks for reporting.

**Comment by swernerx** [comment]
Okay, there is definitely an issue. Comments are parsed during the actual JavaScript code is parsed. This is mainly because these comments might contain meta information. Parsing comments also means passing the text content of each comment to markdown to create text and HTML representations. Maybe I'll be able to postpone this markdown transformation to the API generator. Not sure about that though as it also means that no plain text representation is available... mhh. Need to think about this a little

**Comment by swernerx** [comment]
Fixed with 0.7.4
